zokrates_core/src/static_analysis/constant_inliner.rs

```diff
@@ -108,7 +100,10 @@ impl<'ast, T: Field> Folder<'ast, T> for ConstantInliner<'ast, T> {
     ) -> FieldElementExpression<'ast, T> {
         match e {
             FieldElementExpression::Identifier(ref id) => match self.get_constant(id) {
-                Some(c) => fold_field_expression(self, c.expression.try_into().unwrap()),
+                Some(c) => {
+                    let c = self.fold_constant(c);
+                    fold_field_expression(self, c.try_into().unwrap())
```

Comment:
Unless I'm missing something, folding isnt required here, it was done in the previous line.

Reply:
this could even be

```
Some(c) => self.fold_constant(c).try_into().unwrap()
```